In [3]:
import numpy as np

In [4]:
y_true = np.array([110, 135, 90, 89, 101, 145, 177, 90, 88, 105])
y_pred = np.array([105, 110, 87, 94, 98, 122, 150, 89, 84, 102])

In [5]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

Cреднеквадратическая ошибка

In [6]:
mean_squared_error(y_true, y_pred)

197.7

Корень среднеквадратической ошибки

In [7]:
mean_squared_error(y_true, y_pred, squared=False)

14.060583202698243

Средний модуль ошибки

In [8]:
mean_absolute_error(y_true, y_pred)

9.9

Коэффициент детерминации

In [9]:
r2_score(y_true, y_pred)

0.7553217821782179

In [11]:
y_true.mean()

113.0

### Вывод

R2 равен 0.755. Это означает, что модель объясняет 75,5% дисперсии целевой переменной. Это – достаточно хорошая точность.

MAE равен 9.9. С учетом того, что среднее значение целевой переменной – 113, точность является достаточно высокой.

RMSE равно 14 и почти в полтора раза больше, чем MAE. Это говорит, что среди значений ошибок есть выбросы.

Таким образом, модель имеет достаточно хорошую точность.